src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -101,6 +104,45 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
             });
         }
 
+        /// <summary>
+        /// Iterate through all function secret files and remove any that don't correspond
+        /// to a function.
+        /// </summary>
+        /// <param name="rootScriptPath">The root function directory.</param>
+        public void PurgeOldFiles(string rootScriptPath)
+        {
+            if (!Directory.Exists(rootScriptPath))
+            {
+                return;
+            }
+
+            var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
+            var secretsDirectory = new DirectoryInfo(_secretsPath);
+
+            foreach (var secretFile in secretsDirectory.GetFiles("*.json"))
+            {
+                string fileName = Path.GetFileNameWithoutExtension(secretFile.Name);
+                if (string.Compare(fileName, "host") == 0)
```

Comment:
I've added a new validation check that throws an error if a function is named "Host". We don't allow that now. Also added a constant for the name.